src/System.Private.CoreLib/src/System/Reflection/Runtime/Assemblies/AssemblyNameHelpers.cs

```diff
@@ -32,8 +32,18 @@ public static String ComputeDisplayName(RuntimeAssemblyName a)
 
             if (a.Version != null)
             {
-                sb.Append(", Version=");
-                sb.Append(a.Version.ToString());
+                Version canonicalizedVersion = a.Version.CanonicalizeVersion();
+                if (canonicalizedVersion.Major != ushort.MaxValue)
```

Comment:
Nit: do we want a named constant for the `ushort.MaxValue` to attach a semantic to it? It's easier to search the codebase for those.